paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkTable.java

```diff
@@ -137,4 +153,97 @@ public Map<String, String> properties() {
             return Collections.emptyMap();
         }
     }
+
+    @Override
+    public StructType partitionSchema() {
+        List<String> partitionKeys = table.partitionKeys();
+        RowType rowType =
+                new RowType(
+                        table.rowType().getFields().stream()
+                                .filter(dataField -> partitionKeys.contains(dataField.name()))
+                                .collect(Collectors.toList()));
+        return SparkTypeUtils.fromPaimonRowType(rowType);
+    }
+
+    @Override
+    public void createPartition(InternalRow internalRow, Map<String, String> map)
+            throws PartitionsAlreadyExistException, UnsupportedOperationException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean dropPartition(InternalRow internalRow) {
+        StructType structType = partitionSchema();
+        StructField[] fields = structType.fields();
+        HashMap<String, String> partitions = new HashMap<>();
```

Comment:
Can you use `RowDataPartitionComputer` here?

And we can convert internalRow to paimon `SparkRow`.

Reply:
Do you mean it like this?
```
        List<String> partitionKeys = table.partitionKeys();
        RowType rowType =
                new RowType(
                        table.rowType().getFields().stream()
                                .filter(dataField -> partitionKeys.contains(dataField.name()))
                                .collect(Collectors.toList()));
        StructType structType = SparkTypeUtils.fromPaimonRowType(rowType);
        String[] partitionCols = table.partitionKeys().toArray(new String[partitionKeys.size()]);
        RowDataPartitionComputer rowDataPartitionComputer = new RowDataPartitionComputer(FileStorePathFactory.PARTITION_DEFAULT_NAME.defaultValue(), rowType, partitionCols);
        LinkedHashMap<String, String> partitions = rowDataPartitionComputer.generatePartValues(new SparkRow(rowType, Row.fromSeq(internalRow.toSeq(structType))));
```
`internalRow` need convert to `Row` ,use `SparkRow` to wrapper this